### PR TITLE
print vault error message on authentication failure

### LINKED ISSUE #273: [FEAT] Return more information on 400 error code
**Is your feature request related to a problem? Please describe.**
Ideally, on error, return more information as the Vault CLI does.

I failed to give a valid role (role didn't exist) during a JWT authentication, but the error message from the Action wasn't clear enough to help me realise my mistake.

e.g. The action returns:

```
Error: Response code 400 (Bad Request)
```

To troubleshoot, I had to ~hack~ upload the JWT to GitHub Actions artifacts, download to my computer, and use the JWT with the Vault CLI to realise my mistake.

**Describe the solution you'd like**
On error return more information as the Vault CLI does.

e.g. on 400, the Vault CLI returns:

```
Code: 400. Errors:

* role "doesnotexist" could not be found
```

**Describe alternatives you've considered**
N/A

**Additional context**
Upon giving the Action an non-existent role for a JWT authentication, the action simply returns:

```
Error: Response code 400 (Bad Request)
```

When attempting the same action via the Vault CLI, the CLI lets me know like this:

```
Error writing data to auth/github/login: Error making API request.

URL: PUT https://vault-dev.example.org/v1/auth/github/login
Code: 400. Errors:

* role "doesnotexist" could not be found
```

The configuration of the action was like so (except I've replaced the real values of `url` and `role`):

```yaml
      - name: Retrieve secret from Vault (dev)
        uses: hashicorp/vault-action@v2.4.0
        with:
          url: https://vault-dev.example.org/
          role: doesnotexist
          method: jwt
          path: github
          secrets: |
              /auth/token/lookup-self display_name | NAME
```

### PR DESCRIPTION
**Pull request by kevinschoonover**
image

similar to #276, but you dont need to enable debug logging

closes #273 

also changes ncc to vercel/ncc

### CODE REVIEW

File: src/auth.js [hunk 1]
```diff
@@ -109,7 +110,16 @@ async function getClientToken(client, method, path, payload) {
     core.debug(`Retrieving Vault Token from v1/auth/${path}/login endpoint`);
 
     /** @type {import('got').Response<VaultLoginResponse>} */
-    const response = await client.post(`v1/auth/${path}/login`, options);
+    let response;
+    try {
+        response = await client.post(`v1/auth/${path}/login`, options);
+    } catch (err) {
+        if (err instanceof got.HTTPError) {
+            throw Error(`failed to retrieve vault token. code: ${err.code}, message ${err.message}, vaultResponse: ${JSON.stringify(err.response.body)}`)
```
Comment (austingebauer) [inline src/auth.js:118]:
```suggestion
            throw Error(`failed to retrieve vault token. code: ${err.code}, message: ${err.message}, vaultResponse: ${JSON.stringify(err.response.body)}`)
```

**Review by austingebauer — APPROVED** [review]
One small suggestion. Otherwise, LGTM! Thanks for this.

**Comment by kevinschoonover** [comment]
@austingebauer fixed the merge conflicts / comments so this should now be good to go

**Comment by austingebauer** [comment]
Thanks, @kevinschoonover!

**Comment by ps-jay** [comment]
Thanks folks!  Quality of life improved ☺️

**Pull request merged**